File: eidocell-backend/schemas/features.py
from pydantic import BaseModel, Field


class RunFeatureExtractionRequest(BaseModel):
    method: str = "morphological"


class FeatureExtractionResult(BaseModel):
    processed: int
    skipped: int
    total: int
    feature_dim: int


class FeatureExtractionMethod(BaseModel):
    id: str
    name: str
    feature_dim: int


class RunDimReductionRequest(BaseModel):
    method: str = "pca"
    n_components: int = Field(2, ge=2, le=3)
    params: dict = {}


class DimReductionResult(BaseModel):
    method: str
    n_components: int
    n_samples: int
    embeddings: list[dict]

File: eidocell-ui/src/types/features.ts
export interface FeatureExtractionMethod {
  id: string
  name: string
  feature_dim: number
}

export interface RunFeatureExtractionRequest {
  method?: string
}

export interface FeatureExtractionResult {
  processed: number
  skipped: number
  total: number
  feature_dim: number
}

export interface RunDimReductionRequest {
  method?: string
  n_components?: number
  params?: Record<string, unknown>
}

export interface EmbeddingPoint {
  sample_id: string
  filename: string
  x: number
  y: number
  z?: number
  class_id: string | null
  class_name: string | null
  class_color: string | null
}

export interface DimReductionResult {
  method: string
  n_components: number
  n_samples: number
  embeddings: EmbeddingPoint[]
}
